internal/ingest/ingest.go

```diff
@@ -293,9 +303,9 @@ func (ing *Ingester) Sync(ctx context.Context, peerID peer.ID, peerAddr multiadd
 		log.Debugw("Syncing advertisements up to latest", "adCid", c)
 		for {
 			select {
-			case adCid := <-syncDone:
-				log.Debugw("Synced advertisement", "adCid", adCid)
-				if adCid == c {
+			case adProcessedEvent := <-syncDone:
+				log.Debugw("Synced advertisement", "adCid", adProcessedEvent.adCid)
+				if adProcessedEvent.adCid == c || adProcessedEvent.err != nil && adProcessedEvent.headAdCid == c {
```

Comment:
Please add comment to explain why `adProcessedEvent.adCid` will not equal `c` on error.